processor/src/tests/debug.rs

```diff
@@ -0,0 +1,365 @@
+    ├── 0x00000002: 3
+    ├── 0x00000003: 0
+    ├── 0x00000004: 5
+    └── 0x00000005: 0
+    Memory state before step 15 for the context 0 in the interval [12, 14]:
+    ├── 0x0000000c: EMPTY
+    ├── 0x0000000d: EMPTY
+    └── 0x0000000e: EMPTY
+    ");
+}
+
+#[test]
+fn test_debug_local() {
+    let stack_inputs = [5, 3, 2, 1];
+    let local_addr_4 = 0x40000006;
+    let source = format!(
+        "
+        proc.test.6
+            # Get address of third local to show it has been initialized to an arbitrary value
+            locaddr.4 push.{local_addr_4} assert_eq
+
+            loc_store.0 loc_store.1 loc_store.2 # => [0->1, 1->2, 2->3]
+
+            debug.mem
+
+            debug.local
+            debug.local.2   # => 3
+            debug.local.4   # => 42 (garbage)
+            debug.local.1.2 # => [2, 3]
+
+
+            loc_store.4 # Overwrite previous garbage value => [4->5]
+            debug.local
+            debug.local.4   # => 5 (not garbage)
+            debug.local.6   # => EMPTY (out of bounds)
+            debug.local.2.6 # => [3, 0, 5, 0, EMPTY]
+        end
+        begin
+            # Write a garbage value to show it is not 0 by default
+            push.42 mem_store.{local_addr_4}
+            exec.test
+            debug.mem
+        end"
+    );
+
+    // Execute with debug buffer
+    let test = build_debug_test!(source, &stack_inputs);
+    let (_stack, output) = test.execute_with_debug_buffer().expect("execution failed");
+
+    insta::assert_snapshot!(output, @r"
+    Memory state before step 25 for the context 0:
+    ├── 0x40000000: 0
+    ├── 0x40000001: 0
+    ├── 0x40000002: 1
+    ├── 0x40000003: 2
+    ├── 0x40000004: 3
+    ├── 0x40000005: 0
+    ├── 0x40000006: 42
+    └── 0x40000007: 0
+    State of procedure locals [0, 5] before step 25:
+    ├──     0: 1
+    ├──     1: 2
+    ├──     2: 3
+    ├──     3: 0
+    ├──     4: 42
+    └──     5: 0
+    State of procedure local 2 before step 25: 3
+    State of procedure local 4 before step 25: 42
+    State of procedure locals [1, 2] before step 25:
+    ├──     1: 2
+    └──     2: 3
+    State of procedure locals [0, 5] before step 29:
+    ├──     0: 1
+    ├──     1: 2
+    ├──     2: 3
+    ├──     3: 0
+    ├──     4: 5
+    └──     5: 0
+    State of procedure local 4 before step 29: 5
+    State of procedure local 6 before step 29: EMPTY
+    State of procedure locals [2, 6] before step 29:
+    ├──     2: 3
+    ├──     3: 0
+    ├──     4: 5
+    ├──     5: 0
+    └──     6: EMPTY
+    Memory state before step 34 for the context 0:
+    ├── 0x40000000: 0
+    ├── 0x40000001: 0
+    ├── 0x40000002: 1
+    ├── 0x40000003: 2
+    ├── 0x40000004: 3
+    ├── 0x40000005: 0
+    ├── 0x40000006: 5
+    └── 0x40000007: 0
+    ");
+}
+
+#[test]
+fn test_debug_adv_stack() {
+    let stack_input = [0, 1, 2, 3, 4]; // 4 is on top
+    let advice_stack = [8, 7, 6, 5, 4, 3, 2, 1]; // 8 is on top
```

Comment:
Not related to this PR, but it is definitely confusing that we specify operand stack inputs in one way and advice stack inputs in another way. I think maybe the fast processor fixed this already?